depth_image_proc/include/depth_image_proc/conversions.hpp

```diff
@@ -65,19 +65,27 @@ void convertDepth(
   float constant_y = unit_scaling / model.fy();
   float bad_point = std::numeric_limits<float>::quiet_NaN();
 
+  // ensure that the computation only happens in case we have a default depth
+  T invalid_depth_cvt = T(0);
+  bool use_invalid_depth = invalid_depth != 0.0;
+  if (use_invalid_depth) {
+    invalid_depth_cvt = DepthTraits<T>::fromMeters(invalid_depth);
+  }
   sensor_msgs::PointCloud2Iterator<float> iter_x(*cloud_msg, "x");
   sensor_msgs::PointCloud2Iterator<float> iter_y(*cloud_msg, "y");
   sensor_msgs::PointCloud2Iterator<float> iter_z(*cloud_msg, "z");
+
+  // TODO(philipppolterauer): i think this is undefined behaviour we should favor memcpy? https://stackoverflow.com/questions/55150001/vector-with-reinterpret-cast
```

Comment:
this isn't undefined behavior - the PointCloud2Iterator is actually designed for exactly this use case - it computes what the offset is for each of the x/y/z elements inside the point cloud buffer

Reply:
Let's remove the comment - but file a ticket to investigate

Reply:
```suggestion
```